src/main/java/org/folio/circulation/domain/representations/RequestByInstanceIdRequest.java

```diff
@@ -0,0 +1,86 @@
+package org.folio.circulation.domain.representations;
+
+import static org.folio.circulation.support.JsonPropertyFetcher.getDateTimeProperty;
+import static org.folio.circulation.support.JsonPropertyFetcher.getUUIDProperty;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ValidationErrorFailure.failedValidation;
+
+import java.util.UUID;
+
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+
+import io.vertx.core.json.JsonObject;
+
+public class RequestByInstanceIdRequest {
+  private static final String REQUEST_DATE = "requestDate";
+  private static final String REQUESTER_ID = "requesterId";
+  private static final String INSTANCE_ID = "instanceId";
+  private static final String REQUEST_EXPIRATION_DATE = "requestExpirationDate";
+  private static final String PICKUP_SERVICE_POINT_ID = "pickupServicePointId";
+
+  private final DateTime requestDate;
+  private final UUID requesterId;
+  private final UUID instanceId;
+  private final DateTime requestExpirationDate;
+  private final UUID pickupServicePointId;
+
+  private RequestByInstanceIdRequest(
+      DateTime requestDate,
+      UUID requesterId,
+      UUID instanceId,
+      DateTime requestExpirationDate,
+      UUID pickupServicePointId) {
+    this.requestDate = requestDate;
+    this.requesterId = requesterId;
+    this.instanceId = instanceId;
+    this.requestExpirationDate = requestExpirationDate;
+    this.pickupServicePointId = pickupServicePointId;
+  }
+
+  public static Result<RequestByInstanceIdRequest> from(JsonObject json) {
+    final DateTime requestDate = getDateTimeProperty(json, REQUEST_DATE);
+
+    if (requestDate == null) {
```

Comment:
Please can tests be added to demonstrate the validation for request date and requester ID

Reply:
I'll write unit tests for it. The existing API test was merely to check that a failed validation will fail the request gracefully.